### PR TITLE
[4.2] Consider permissions for MM actions

### PR DESCRIPTION
**Pull request by laoneo**
Pull Request for Issue #36549.

### Summary of Changes
Passes the global permissions from the media manager to the client app. The item can define a flag as well to specify if the item itself can be edited/deleted. This allows external adapters to define also per item permissions.

This pr is based on the pr from @dgrammatiko and #36556.

### Testing Instructions
- Check the instructions in issue #36549.
- Additionally add a canEdit/canDelete flag to the file /plugins/filesystem/local/src/Adapter/LocalAdapter.php on line 344.

### Actual result BEFORE applying this Pull Request
Actions are always shown.

### Expected result AFTER applying this Pull Request
Actions are only shown when the permissions are correct.

### CODE REVIEW

**Comment by Quy** [comment]
Delete permission is denied.
No delete button, however, there is a delete item action.

36932-delete

**Comment by laoneo** [comment]
Did you do `npm ci`? As this is something which is handled in the  `actionItemsContainer.vue` file.

**Comment by Quy** [comment]
Tested with the prebuilt package so no need to npm ci.

**Comment by laoneo** [comment]
Was a copy paste error, can you test again?

**Comment by Quy** [comment]
Will test when I get into the office later today. Thanks.

File: administrator/components/com_media/resources/scripts/store/actions.es6.js [hunk 1]
```diff
@@ -172,15 +178,22 @@ export const uploadFile = (context, payload) => {
...
- * @param payload object: the old and the new path
+ * @param payload object: the item and the new path
  */
 export const renameItem = (context, payload) => {
+  if (!api.canEdit) {
+    return;
+  }
+
+  if (typeof payload.item.canEdit !== 'undefined' && payload.item.canEdit === false) {
+	return;
```
Comment (Quy) [inline administrator/components/com_media/resources/scripts/store/actions.es6.js]:
```suggestion
	  return;
```

File: administrator/components/com_media/resources/scripts/components/browser/items/document.vue [hunk 2]
```diff
@@ -23,8 +23,6 @@
       ref="container"
       :focused="focused"
       :item="item"
-      :edit="editItem"
```
Comment (Quy) [inline administrator/components/com_media/resources/scripts/components/browser/items/document.vue:26]: Delete lines 52-53.

Comment (laoneo) [reply administrator/components/com_media/resources/scripts/components/browser/items/document.vue:26]: DOne

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 95b4f88ef3777c602c2a6c2f43561aab02ae4baa<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36932">issues.joomla.org/tracker/joomla-cms/36932</a>.</sub>

**Comment by brianteeman** [comment]
Don't know if this is beyond the scope or not. But when you have edit permission but there are no media plugins enabled the pencil icon is still displayed

**Comment by laoneo** [comment]
> Don't know if this is beyond the scope or not. But when you have edit permission but there are no media plugins enabled the pencil icon is still displayed

I would handle this is in it's own pr.

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 29fd7f884292dba158b6a1f87539fa7bc65a56f7<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36932">issues.joomla.org/tracker/joomla-cms/36932</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36932">issues.joomla.org/tracker/joomla-cms/36932</a>.</sub>

**Comment by roland-d** [comment]
@laoneo Can you please check the conflicts?

**Comment by laoneo** [comment]
Done

**Comment by roland-d** [comment]
Thanks everybody.

**Pull request merged**